sdk/src/main/scala/com/horizen/account/state/StateDbAccountStateView.scala

```diff
@@ -178,6 +178,14 @@ class StateDbAccountStateView(
       throw new IllegalArgumentException(s"Unsupported transaction type ${tx.getClass.getName}")
 
     val ethTx = tx.asInstanceOf[EthereumTransaction]
+
+    // should never happen if the tx has been accepted in mempool.
+    // In some negative test scenario this can happen when forcing an unsigned tx to be forged in a block.
+    // In this case the 'from' attribute in the msg would not be 
+    // set, and it would be difficult to rootcause the reason why gas and nonce checks would fail
+    if (!ethTx.isSigned)
+      throw new IllegalArgumentException(s"Transaction is not signed: ${ethTx.id}")
```

Comment:
don't we check it when validate the block during the history apply as a part of ChainIdBlockSemanticValidator?

Reply:
Here we are forging a block with a given list of transactions got from mempool or forced by a specific test, the block semantic validation takes place afterwards, but then the offending tx would not have been included in the resulting block.

Reply:
ok, got it